### PR TITLE
Feature/summary log dev

### PR DESCRIPTION
**Pull request by seth127**
I have been working on this with the Matt R and 10am group and @aspirinpkpd (Ana R) specifically.

There are some ongoing tweaks that are being made, but I would like to merge this to `develop` before it becomes _too_ monstrous (maybe too late?) and then make those tweaks on additional branches. 

The functionality outlined in https://github.com/metrumresearchgroup/rbabylon/issues/53 and https://github.com/metrumresearchgroup/rbabylon/issues/67 is covered in this PR.

### CODE REVIEW

**Comment by seth127** [comment]
@dpastoor @kylebaron @copernican You three seem like the most logical candidates to review this, although it could be Ana R or Adam L too. Any thoughts?

**Comment by seth127** [comment]
Discussed with Kyle and Sean and @copernican is going to take the code review.

File: tests/testthat/test-model-summaries.R [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  invisible(copy_model_from(yaml_ext(MOD1_PATH), NEW_MOD2, "model from test-model-summaries.R", .directory = "."))
+  invisible(copy_model_from(yaml_ext(MOD1_PATH), NEW_MOD3, "model from test-model-summaries.R", .directory = "."))
+  fs::dir_copy(MOD1_PATH, NEW_MOD2)
+  fs::dir_copy(MOD1_PATH, NEW_MOD3)
+})
+teardown({
+  cleanup()
+})
+
+withr::with_options(list(rbabylon.bbi_exe_path = '/data/apps/bbi',
```
Comment (copernican) [inline tests/testthat/test-model-summaries.R:38]: The absolute path should be replaced with `read_bbi_path()`.

File: tests/testthat/test-summary-log.R [hunk 2]
```diff
@@ -0,0 +1,100 @@
+context("Test creating summary logs")
+
+if (Sys.getenv("METWORX_VERSION") == "" && Sys.getenv("DRONE") != "true") {
+  skip("test-summary only runs on Metworx or Drone")
+}
+
+source("data/test-workflow-ref.R")
```
Comment (copernican) [inline tests/testthat/test-summary-log.R:7]: We should change from the `source()` pattern to the `testthat` handling for special files, as described in metrumresearchgroup/mrgvalidatetestreference#8 (at least for all new development, if we don't want to change everything right now).

File: tests/testthat/test-config-log.R [hunk 3]
```diff
@@ -0,0 +1,94 @@
+context("Constructing config log from bbi_config.json")
+
+source("data/test-workflow-ref.R")
```
Comment (copernican) [inline tests/testthat/test-config-log.R:3]: We should change from the `source()` pattern to the `testthat` handling for special files, as described in metrumresearchgroup/mrgvalidatetestreference#8 (at least for all new development, if we don't want to change everything right now).

File: tests/testthat/test-model-summaries.R [hunk 4]
```diff
@@ -0,0 +1,82 @@
+context("Test bbi summary on multiple models")
+
+if (Sys.getenv("METWORX_VERSION") == "" && Sys.getenv("DRONE") != "true") {
+  skip("test-summary only runs on Metworx or Drone")
+}
+
+source("data/test-workflow-ref.R")
```
Comment (copernican) [inline tests/testthat/test-model-summaries.R:7]: We should change from the `source()` pattern to the `testthat` handling for special files, as described in metrumresearchgroup/mrgvalidatetestreference#8 (at least for all new development, if we don't want to change everything right now).

File: tests/testthat/test-run-log.R [hunk 5]
```diff
@@ -2,9 +2,8 @@ context("Constructing run log from model yaml")
 
 source("data/test-workflow-ref.R")
```
Comment (copernican) [inline tests/testthat/test-run-log.R:3]: We should change from the `source()` pattern to the `testthat` handling for special files, as described in metrumresearchgroup/mrgvalidatetestreference#8 (at least for all new development, if we don't want to change everything right now).

File: tests/testthat/test-summary-log.R [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  })
+
+  test_that("summary_log() returns NULL and warns when no YAML found", {
+    log_df <- expect_warning(summary_log("data"), regexp = "Found no valid model YAML files in data")
+    expect_true(is.null(log_df))
+  })
+}) # closing withr::with_options
+
+
+withr::with_options(list(rbabylon.bbi_exe_path = '/data/apps/bbi',
```
Comment (copernican) [inline tests/testthat/test-summary-log.R:53]: The absolute path should be replaced with `read_bbi_path()`.

**Comment by seth127** [comment]
@copernican I made all the changes and refactored the tests to use `setup-` instead of sourcing files. Back to you for approval.

**Comment by seth127** [comment]
In consultation with @copernican we've decided to abandon this branch and copy the relevant functionality to two separate branches for easier review and a more sensible commit history. See the following PR's:

* https://github.com/metrumresearchgroup/rbabylon/pull/117
* https://github.com/metrumresearchgroup/rbabylon/pull/118

**Pull request closed without merging**